pkg/kvstore/redis.go

```diff
@@ -162,13 +169,50 @@ func (s *redisKvStore) PutBatch(ctx context.Context, values interface{}) error {
 		return fmt.Errorf("could not convert values from %T to map[interface{}]interface{}", values)
 	}
 
+	chunkSize := s.settings.BatchSize
+	pairs := make([]interface{}, 0, chunkSize)
 	for k, v := range mii {
-		if err = s.Put(ctx, k, v); err != nil {
-			return fmt.Errorf("failed to write batch to redis: %w", err)
+		key, value, err := s.MarshalKeyValue(k, v)
+		if err != nil {
+			return fmt.Errorf("PutBatch could not marshal key/value: %w", err)
+		}
+		pairs = append(pairs, key)
```

Comment:
I think this can be written as `pairs = append(pairs, key, value)`, making it a little bit more compact

Reply:
done